### DISCUSSION TITLE
Better error message, with stack

### DISCUSSION THREAD

**Discussion by amunhoz** [uNetworking/uWebSockets.js#276] [Q&A] [answered]
Hi,
First, congratulations for the project.  We need something like this to improve node.js performance.

Second, is that possible to generate better error messages?  
Its been a little hard to find where the errors ocurrs since there is not a stack information in the error message.
An object property with the current socket status could help too.

Thanks

**Comment by dalisoft** [comment]
@amunhoz I sure Try/catch will help you faster and better

**Comment by amunhoz** [comment]
The problem is, somethimes there is a lot of code in the try/catch, and when an error occurs, i have to put breakpoints in the entire code to discover where the problem is.
Besides that, is a commom pattern for nodejs errors to put the stack info.

For now, i've made a wrapper to replace the original functions with other ones to give me more info about the error and its location.

But i think this can be implemented in the package.

**Comment by ghost** [comment]
Hello,

I don't follow, could you give an example? You shouldn't get "errors" at all, you only have callbacks and what they provide. Status never change unless a callback was called to signal that change.

**Comment by amunhoz** [comment]
Sorry, i think this explains better:

```
Try {
   console.logx("lalala")
} catch (e) {
//error
}
```

The error instance will have two main informations:

**message**:"console.logx is not a function"
**stack**:"TypeError: console.logx is not a function\n    at Object.<anonymous> (/home/ivuser/git/_test/testrepo/index.js:2:13)\n    at Module._compile (internal/modules/cjs/loader.js:775:14)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)\n    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)\n    at startup (internal/bootstrap/node.js:283:19)\n    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)"



The stack information help to find where the exception was throwed.
Right now, uWebsockets.js give only a text with the error message.

**Comment by ghost** [comment, marked as answer]
I have a **very** hard time understanding how things could be any more clear:

```
/home/alexhultman/HelloWorld.js:14
consoles.logz();
^

ReferenceError: consoles is not defined
    at /home/alexhultman/HelloWorld.js:14:1
```

This **is** the stack trace. It is very shallow because, well, the stack **is** this shallow at this particular point.

**Comment by amunhoz** [comment]
Your example show how it supose to be... 
However, like i've said: uWebsockets.js is not giving the stack trace information in the event of an error.   Only a text information.

**Comment by ghost** [comment]
To make sure I wasn't wrong on this, I made yet another test:

```
/home/alexhultman/uWebSockets.js/examples/HelloWorld.js:11
        app.publish();
            ^
Function requires at least 2 arguments.
```

This clearly shows the stack trace (which at this point is only 1 deep) and a clear V8 exception.
